### ISSUE TITLE
pacman detects the "beta -> final release" transition as a downgrade

### ISSUE THREAD

**Issue by yakubin** [holocm/holo-build#25]
I went over the `1.0.0-alpha.1 < 1.0.0-alpha.3 < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0` sequence again to double-check whether using "." as the separator (introduced in holocm/libpackagebuild@b0abdf3 imported by c004bbd) solves the issue with "_" which I described in the description of 1e825f9. It turns out that no, using "." causes pacman to detect installing `1.0.0` after `1.0.0-beta.11` as a downgrade.

Example pacman output (just the top lines):
```
[jakub@arch-virt pkgs]$ sudo pacman -U 1.0.0/testpkg-1.0.0-2-any.pkg.tar.xz 
loading packages...
warning: downgrading package testpkg (1.0.0.beta11-2 => 1.0.0-2)
```

**Comment by yakubin** [comment]
Reverting  holocm/libpackagebuild@b0abdf3 should solve the issue.

**Comment by yakubin** [comment]
Adding another "." (e.g. `1.0.0.beta.11-2`) doesn't fix the issue:
```
warning: downgrading package testpkg (1.0.0.beta.11-2 => 1.0.0-2)
```

I double-checked and reverting holocm/libpackagebuild@b0abdf3 does solve the issue.

**Comment by yakubin** [comment]
From the `vercmp(8)` manpage on Arch Linux:

```
Version comparison operates as follows:

    Alphanumeric:
     1.0a < 1.0b < 1.0beta < 1.0p < 1.0pre < 1.0rc < 1.0 < 1.0.a < 1.0.1
    Numeric:
     1 < 1.0 < 1.1 < 1.1.1 < 1.2 < 2.0 < 3.0.0
```

**Comment by majewsky** [comment]
Sorry for making such a mess. I should have followed the argument instead of my gut. Before I tag the bugfix release, can you please have a quick look if 7f0e28e produces the correct results for you?

**Comment by yakubin** [comment]
> Sorry for making such a mess. I should have followed the argument instead of my gut.

No worries. Everybody makes mistakes.

> Before I tag the bugfix release, can you please have a quick look if 7f0e28e produces the correct results for you?

It does. Thanks!

**Comment by yakubin** [comment]
I fixed some tests after this change: #26

There are still some that fail, but that's probably because I'm using Debian and its version of `file(1)` may be different, so there is a bunch failing along the lines of:

```
    -foo/bar/package_1.0-1_all.deb: Debian binary package (format 2.0), with control.tar.gz, data compression xz  
    +foo/bar/package_1.0-1_all.deb: Debian binary package (format 2.0)
```
